### ISSUE TITLE
inferSchema with sample data?

### ISSUE THREAD

**Issue by stati** [databricks/spark-csv#139]
inferSchema doc says "It requires one extra pass over the data"
Is there an option to infer schema with a small sample and fail if rest of data doesn't adhere to inferred schema?

**Comment by falaki** [comment]
Not yet. That may come later, especially if users find it very useful in practice. Note that, when inferring schema from a sample, there is a non-zero chance for wrong types or missing fields in the inferred schema.

**Comment by stati** [comment]
Yeah, there is a scope to infer schema incorrectly. Current behavior can be left as default. Quite a few CSVs I'm looking at won't have that issue.

**Comment by jongyonkim** [comment]
I tried using .options(header='true', inferSchema='true') on the cars.csv dataset but it doesn't seem to have inferred the schema properly. Year should be inferred as integer, not string.

Code:
df = sqlContext.read.format('com.databricks.spark.csv').options(header='true', inferSchema='true').load(LOAD_FILE_PATH)
df.printSchema()

Output:
root
 |-- year: string (nullable = true)
 |-- make: string (nullable = true)
 |-- model: string (nullable = true)
 |-- comment: string (nullable = true)
 |-- blank: string (nullable = true)

Am I missing something?

**Comment by falaki** [comment]
What version of spark-csv are you using?

**Comment by jongyonkim** [comment]
I'm using the Databricks REPL notebook. I didn't have to import anything as it should come pre-loaded. Is this the latest version? How do I check?

**Comment by falaki** [comment]
I see. Spark-csv in Databricks is version 1.1.0 which does not include schema inference. We will update to latest version very soon.

**Comment by ganeshayadiyala** [comment]
It would be very helpful if we have inferschema with sample data.
